src/Dfc.CourseDirectory.Core/Search/AzureCognitiveSearch/AzureCognitiveSearchQueryBuilder.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Azure.Search.Documents;
+using Azure.Search.Documents.Models;
+
+namespace Dfc.CourseDirectory.Core.Search.AzureCognitiveSearch
+{
+    public class AzureCognitiveSearchQueryBuilder
+    {
+        const string FilterDelimiter = "|";
+
+        private string _searchText;
+        private SearchMode? _searchMode;
+        private IList<string> _searchFields;
+        private IList<string> _facets;
+        private IList<string> _filters;
+        private IList<string> _orderBy;
+        private string _scoringProfile;
+        private int? _size;
+        private int? _skip;
+        private bool? _includeTotalCount;
+
+        public AzureCognitiveSearchQueryBuilder(string searchText)
+        {
+            _searchText = searchText;
+            _searchFields = new List<string>();
+            _facets = new List<string>();
+            _filters = new List<string>();
+            _orderBy = new List<string>();
+        }
+
+        public AzureCognitiveSearchQueryBuilder WithSearchMode(SearchMode? searchMode)
+        {
+            _searchMode = searchMode;
+            return this;
+        }
+
+        public AzureCognitiveSearchQueryBuilder WithSearchFields(params string[] searchFields)
+        {
+            foreach (var searchField in searchFields)
+            {
+                _searchFields.Add(searchField);
+            }
+            
+            return this;
+        }
+
+        public AzureCognitiveSearchQueryBuilder WithFacets(params string[] facets)
+        {
+            foreach (var facet in facets)
+            {
+                _facets.Add(facet);
+            }
+
+            return this;
+        }
+
+        public AzureCognitiveSearchQueryBuilder WithSearchInFilter(string variable, IEnumerable<string> values)
+        {
+            if (string.IsNullOrWhiteSpace(variable))
+            {
+                throw new ArgumentNullException(nameof(variable));
+            }
+
+            if (values?.Any() ?? false)
+            {
+                _filters.Add($"search.in({variable}, {SearchFilter.Create($"{string.Join(FilterDelimiter, values)}")}, '{FilterDelimiter}')");
```

Comment:
Do `values` need to be escaped here?

Reply:
They are being escaped by the `SearchFilter.Create(...)` method, whether that should be done here or not is another question...?